pkg/provider/containers.go

```diff
@@ -62,6 +69,51 @@ func (c *Container) GetUID() (string, error) {
 	return uid, nil
 }
 
+func (c *Container) SetPreflightResults(preflightImageCache map[string]plibRuntime.Results, env *TestEnvironment) error {
+	// Short circuit if the image already exists in the cache
+	if _, exists := preflightImageCache[c.Image]; exists {
+		c.PreflightResults = preflightImageCache[c.Image]
+		return nil
+	}
+
+	var results plibRuntime.Results
+	opts := []plibContainer.Option{}
+
+	// Check to make sure that the environment variable is set
+	if len(env.GetDockerConfigFile()) > 0 {
+		opts = append(opts, plibContainer.WithDockerConfigJSONFromFile(env.GetDockerConfigFile()))
+	} else {
+		logrus.Errorf("Container func SetPreflightResults has failed due to missing PFLT_DOCKERCONFIG environment variable")
+		return nil
+	}
+	if env.IsPreflightInsecureAllowed() {
+		logrus.Info("Insecure connections are being allowed to preflight")
+		opts = append(opts, plibContainer.WithInsecureConnection())
+	}
+
+	check := plibContainer.NewCheck(c.Image, opts...)
+
+	// Create artifacts handler
+	artifactsWriter, err := artifacts.NewMapWriter()
+	if err != nil {
+		return err
+	}
+	ctx := artifacts.ContextWithWriter(context.Background(), artifactsWriter)
```

Comment:
Configure a logger  to retrieve the logs from preflights. I tried to run it but it only displayed the info logs (V(0)), not sure how to set the level (see https://github.com/redhat-openshift-ecosystem/openshift-preflight/blob/main/docs/LIBRARY.md ):
		
// [2] Configure a logr.Logger compliant logger of your choice, and add it to the context
		logbytes := bytes.NewBuffer([]byte{})
		checklogger := log.Default()
		checklogger.SetOutput(logbytes)
		logger := stdr.New(checklogger)
		ctx = logr.NewContext(ctx, logger.V(3))

	// [4] Accessing logs
	fmt.Println("The preflight log contained:")
	fmt.Println(logbytes.String())

Reply:
Good find, I didn't know this was exposed.  I'll play with adding this.

Reply:
DEBUG  [Feb 14 11:46:48.910][suite.go: 73] Running preflight container tests for: jack  
INFO   [Feb 14 11:46:48.910][containers.go: 77] Running preflight container test against image: quay.io/testnetworkfunction/cnf-test-partner:latest with name: jack 
INFO   [Feb 14 11:47:32.729][containers.go: 125] 2023/02/14 11:47:18 "level"=0 "msg"="check completed" "result"="PASSED" "check"="HasLicense"
2023/02/14 11:47:18 "level"=0 "msg"="check completed" "result"="PASSED" "check"="HasUniqueTag"
2023/02/14 11:47:18 "level"=0 "msg"="check completed" "result"="PASSED" "check"="LayerCountAcceptable"
2023/02/14 11:47:18 "level"=0 "msg"="check completed" "result"="PASSED" "check"="HasNoProhibitedPackages"
2023/02/14 11:47:18 "level"=0 "msg"="check completed" "result"="PASSED" "check"="HasRequiredLabel"
2023/02/14 11:47:18 "level"=0 "msg"="USER tnf-user specified that is non-root"
2023/02/14 11:47:18 "level"=0 "msg"="check completed" "result"="PASSED" "check"="RunAsNonRoot"
2023/02/14 11:47:29 "level"=0 "msg"="check completed" "result"="PASSED" "check"="HasModifiedFiles"
2023/02/14 11:47:30 "level"=0 "msg"="check completed" "result"="PASSED" "check"="BasedOnUbi"
2023/02/14 11:47:30 "level"=0 "msg"="This image's tag latest will be paired with digest sha256:61d60fe79642048e8b69e318fbc1616e3c0cbd792aa5573d327b950b0f9e1926 once this image has been published in accordance with Red Hat Certification policy. You may then add or remove any supplemental tags through your Red Hat Connect portal as you see fit."

Reply:
Yeah I'm only seeing what is essentially the bare minimum of logs.  I wonder if in the future we might be able to specify a more verbose log level.

Reply:
See logging lines added in https://github.com/test-network-function/cnf-certification-test/pull/631/commits/f644eddca80dafc0effcb345820eee4f8aaf3c40